### ISSUE TITLE
Null pointer exception when latitude < 0

### ISSUE THREAD

**Issue by okievv** [zdila/gpx-animator#145]
When setting a negative minimum latitude, e.g. -57, I get a null pointer exception. I wanted to reflect full South America. See my config and gpx files attached, release is 1.4.
Regards, okievv

**Excerpt of java stack:**
[  0%] computed zoom is 4
java.util.concurrent.ExecutionException: java.lang.NegativeArraySizeException
	at java.util.concurrent.FutureTask.report(FutureTask.java:122)
	at java.util.concurrent.FutureTask.get(FutureTask.java:192)
	at javax.swing.SwingWorker.get(SwingWorker.java:602)
	at sk.freemap.gpxAnimator.ui.MainFrame$14$1.done(MainFrame.java:449)
	at javax.swing.SwingWorker$5.run(SwingWorker.java:737)
	at javax.swing.SwingWorker$DoSubmitAccumulativeRunnable.run(SwingWorker.java:832)
	at sun.swing.AccumulativeRunnable.run(AccumulativeRunnable.java:112)
	at javax.swing.SwingWorker$DoSubmitAccumulativeRunnable.actionPerformed(SwingWorker.java:842)
	at javax.swing.Timer.fireActionPerformed(Timer.java:313)
	at javax.swing.Timer$DoPostEvent.run(Timer.java:245)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:311)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:756)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.awt.EventQueue$3.run(EventQueue.java:703)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:76)
...
[min-lat-error.zip](https://github.com/zdila/gpx-animator/files/4005254/min-lat-error.zip)

**Comment by McPringle** [comment]
The problem is located in the `render` method which is really large, complicated and hard to understand. Before fixing this issue it makes more sense to refactor that method. For now, it is too risky to break more when trying to fix this issue.

This issue is now blocked by #148.

**Comment by McPringle** [comment]
Happy New Year!

@zdila, maybe you are able to find some time to help to fix this issue. Even after refactoring the `render` method (#148), mostly with *Extract Method* refactorings, my understanding of the calculation of the map area is still very limited. It looks like the problem here is using the negative latitude to point to the south of the equator, which somehow breaks the calculation.

I would be lucky to get some help from you. Of course, not today on New Year's Day, the family has priority... :-)

**Comment by zdila** [comment]
Hi @McPringle and PF 2020.

I can promise nothing. Moreover I think that after so many years I won't be more clever with reading my code than you ;-)

**Comment by McPringle** [comment]
With #238, I fixed a problem in the calculation of the correct width of the video file. It did not fix this problem, but it is a step forward. The exception now looks like (see the `0`):

```
java.lang.IllegalArgumentException: invalid video frame size [792 x 0]
java.util.concurrent.ExecutionException: java.lang.IllegalArgumentException: invalid video frame size [792 x 0]
	at java.base/java.util.concurrent.FutureTask.report(FutureTask.java:122)
	at java.base/java.util.concurrent.FutureTask.get(FutureTask.java:191)
	at java.desktop/javax.swing.SwingWorker.get(SwingWorker.java:613)
	at app.gpx_animator.ui.MainFrame$5.done(MainFrame.java:439)
	at java.desktop/javax.swing.SwingWorker$5.run(SwingWorker.java:750)
	at java.desktop/javax.swing.SwingWorker$DoSubmitAccumulativeRunnable.run(SwingWorker.java:847)
	at java.desktop/sun.swing.AccumulativeRunnable.run(AccumulativeRunnable.java:112)
	at java.desktop/javax.swing.SwingWorker$DoSubmitAccumulativeRunnable.actionPerformed(SwingWorker.java:857)
	at java.desktop/javax.swing.Timer.fireActionPerformed(Timer.java:310)
	at java.desktop/javax.swing.Timer$DoPostEvent.run(Timer.java:242)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:316)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:770)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:715)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:391)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:740)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)
Caused by: java.lang.IllegalArgumentException: invalid video frame size [792 x 0]
	at com.xuggle.mediatool.MediaWriter.addVideoStream(MediaWriter.java:491)
	at com.xuggle.mediatool.MediaWriter.addVideoStream(MediaWriter.java:416)
	at app.gpx_animator.core.renderer.framewriter.VideoFrameWriter.<init>(VideoFrameWriter.java:33)
	at app.gpx_animator.Renderer.render(Renderer.java:169)
	at app.gpx_animator.ui.MainFrame$5.doInBackground(MainFrame.java:407)
	at app.gpx_animator.ui.MainFrame$5.doInBackground(MainFrame.java:404)
	at java.desktop/javax.swing.SwingWorker$1.call(SwingWorker.java:304)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.desktop/javax.swing.SwingWorker.run(SwingWorker.java:343)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
	at java.base/java.lang.Thread.run(Thread.java:832)
```

**Comment by McPringle** [comment]
@okievv After the fix for #238 applied, there is an easy workaround: Specify min and max values, not just the min value. For example, Max Lat = 50 and Min Lat = -57 will render your video fine with version 1.6.

**Comment by McPringle** [comment]
To fix this problem I'll enforce the workaround from above. If you enter a longitude, you have to enter both, min and max. Same with the latitude.
